buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
+							})
+							h.AssertEq(t, br.MetRequires, []string{"some-deprecated-bp-replace-version-dep", "some-dep", "some-replace-version-dep"})
+						})
 
-					it("should error if there are undeclared media types", func() {
-						buildpackID := bpTOML.Buildpack.ID
-						bpTOML.Buildpack.SBOM = []string{"application/vnd.cyclonedx+json"}
+						when("build.toml", func() {
+							it("errors when the build bom has a top level version", func() {
+								h.Mkfile(t,
+									"[[bom]]\n"+
+										`name = "some-dep"`+"\n"+
+										`version = "some-version"`+"\n",
+									filepath.Join(appDir, "build-A-v1.toml"),
+								)
+								_, err := descriptor.Build(buildpack.Plan{}, config, mockEnv)
+								h.AssertNotNil(t, err)
+								expected := "top level version which is not allowed"
+								h.AssertStringContains(t, err.Error(), expected)
+							})
+						})
 
-						h.Mkdir(t,
-							filepath.Join(layersDir, buildpackID))
-						h.Mkfile(t, `{"key": "some-bom-content"}`,
-							filepath.Join(layersDir, buildpackID, "launch.sbom.spdx.json"))
+						when("launch.toml", func() {
+							it("errors when the launch bom has a top level version", func() {
+								h.Mkfile(t,
+									"[[bom]]\n"+
+										`name = "some-dep"`+"\n"+
+										`version = "some-version"`+"\n",
+									filepath.Join(appDir, "launch-A-v1.toml"),
+								)
+								_, err := descriptor.Build(buildpack.Plan{}, config, mockEnv)
+								h.AssertNotNil(t, err)
+								expected := "top level version which is not allowed"
+								h.AssertStringContains(t, err.Error(), expected)
+							})
+						})
 
-						_, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-						h.AssertError(t, err, fmt.Sprintf("validating SBOM file '%s' for buildpack: 'A@v1': undeclared SBOM media type: 'application/spdx+json'", filepath.Join(layersDir, buildpackID, "launch.sbom.spdx.json")))
+						when("SBOM files", func() {
+							it("are ignored", func() {
+								descriptor.API = api.MustParse("0.6").String()
+								buildpackID := descriptor.Buildpack.ID
+								layerName := "some-layer"
+
+								h.Mkdir(t,
+									filepath.Join(layersDir, buildpackID))
+								h.Mkfile(t, `{"key": "some-bom-content"}`,
+									filepath.Join(layersDir, buildpackID, "launch.sbom.cdx.json"),
+									filepath.Join(layersDir, buildpackID, "build.sbom.cdx.json"),
+									filepath.Join(layersDir, buildpackID, fmt.Sprintf("%s.sbom.cdx.json", layerName)))
+
+								h.Mkdir(t,
+									filepath.Join(layersDir, buildpackID, layerName))
+								h.Mkfile(t, "[types]\n  cache = true",
+									filepath.Join(layersDir, buildpackID, fmt.Sprintf("%s.toml", layerName)))
+
+								br, err := descriptor.Build(buildpack.Plan{}, config, mockEnv)
+								h.AssertNil(t, err)
+
+								h.AssertEq(t, len(br.BOMFiles), 0)
+								expected := "the following SBOM files will be ignored for buildpack api version < 0.7"
+								assertLogEntry(t, logHandler, expected)
+							})
+						})
 					})
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L719-L730